java-checks/src/main/java/org/sonar/java/checks/methods/AbstractMethodDetection.java

```diff
@@ -70,6 +76,10 @@ protected void onConstructorFound(NewClassTree newClassTree) {
     // Do nothing by default
   }
 
+  protected void onMethodReferenceFound(MethodReferenceTree methodReferenceTree) {
```

Comment:
From my point of view, this is not yet complete to add support of method references. MethodMatchers are relying on the same mechanism and should support them as well.

I would expect to modify the `MethodMatcher` and `MethodMatcherCollection` classes to add some  `matches(MethodReferenceTree)` / `anyMatch(MethodReferenceTree)` methods there.